### PR TITLE
core: add new base artifact, HostFormFactor

### PR DESCRIPTION
**Pull request by connorjclark**
Extracted from #9911.

Needed for #9713

### CODE REVIEW

File: lighthouse-core/gather/gather-runner.js [hunk 1]
```diff
@@ -456,14 +456,16 @@ class GatherRunner {
 
     const {emulatedFormFactor} = options.settings;
     // Whether Lighthouse was run on a mobile device (i.e. not on a desktop machine).
-    const IsMobileHost = hostUserAgent.includes('Android') || hostUserAgent.includes('Mobile');
+    const HostFormFactor = hostUserAgent.includes('Android') || hostUserAgent.includes('Mobile') ?
+      'mobile' : 'desktop';
     const TestedAsMobileDevice = emulatedFormFactor === 'mobile' ||
```
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: this seems like a good thing to throw into our smoke tests now, do we have any that test as desktop? shouldn't be too bad to add

Comment (connorjclark) [reply lighthouse-core/gather/gather-runner.js]: but this is the `hostUserAgent`. we don't run smoke tests against real mobile devices :P

File: lighthouse-core/gather/gather-runner.js [hunk 2]
```diff
@@ -456,14 +456,15 @@ class GatherRunner {
 
     const {emulatedFormFactor} = options.settings;
     // Whether Lighthouse was run on a mobile device (i.e. not on a desktop machine).
-    const IsMobileHost = hostUserAgent.includes('Android') || hostUserAgent.includes('Mobile');
+    const HostFormFactor = /android|mobile/i.test(hostUserAgent) ? 'mobile' : 'desktop';
```
Comment (connorjclark) [inline lighthouse-core/gather/gather-runner.js]: paul called me names (not allowed, but in his head) for wanting to use a regex here

Comment (paulirish) [reply lighthouse-core/gather/gather-runner.js]: so you want to use a regex. now you got ~[2 problems](https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/)~ 3 problems, including name-calling.

🙃


-----------

case insensitivity is fine. CONFIDENT PROGRAMMING. WE"LL BE FINE.

- https://cs.chromium.org/chromium/src/components/cronet/android/java/src/org/chromium/net/impl/UserAgent.java?l=42&rcl=0efaa7ede88f9448f19193dc513e57d651cabe79
- https://cs.chromium.org/chromium/src/third_party/devtools-frontend/src/node_modules/useragent/lib/regexps.js?l=727&rcl=879d97b05dfe9f26ede820b0296b125d4a68e5ff

surely.

File: types/artifacts.d.ts [hunk 3]
```diff
@@ -29,6 +29,8 @@ declare global {
       LighthouseRunWarnings: string[];
       /** Whether the page was loaded on either a real or emulated mobile device. */
       TestedAsMobileDevice: boolean;
+      /** Device which Chrome is running on. */
+      HostFormFactor: 'desktop'|'mobile';
```
Comment (brendankenny) [inline types/artifacts.d.ts]: `TestedAsMobileDevice`
`HostUserAgent`
`NetworkUserAgent`
`HostFormFactor`
`emulatedFormFactor`
`deviceScreenEmulationMethod`

https://www.theonion.com/taco-bells-five-ingredients-combined-in-totally-new-way-1819564909

File: lighthouse-core/test/gather/gather-runner-test.js [hunk 4]
```diff
@@ -234,6 +234,32 @@ describe('GatherRunner', function() {
...
+          settings: {},
+        });
+        const options = {requestedUrl, driver, settings: config.settings};
+
+        const results = await GatherRunner.run(config.passes, options);
+        expect(results.HostFormFactor).toBe(expectedValue);
+      });
+    }
+
+    test('works when running on mobile device', 'Some Mobile UA', 'mobile');
```
Comment (brendankenny) [inline lighthouse-core/test/gather/gather-runner-test.js]: these would probably be better with some real UA strings

File: lighthouse-core/scripts/build-report-for-autodeployment.js [hunk 5]
```diff
@@ -77,6 +77,7 @@ async function generateErrorLHR() {
       `Something went wrong with recording the trace over your page load. Please run Lighthouse again. (NO_FCP)`, // eslint-disable-line max-len
     ],
     TestedAsMobileDevice: true,
+    HostFormFactor: 'desktop',
```
Comment (brendankenny) [inline lighthouse-core/scripts/build-report-for-autodeployment.js]: should probably also add `HostFormFactor` to the sample artifacts

File: lighthouse-core/gather/gather-runner.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/gather/gather-runner.js]: I know that, notice how this comment is on the `TestedAsMobileDevice` line :P

```
HostFormFactor: 'desktop',
TestedAsMobileDevice: true,
```

seems like a reasonable thing to have in our artifacts smoketests

**Pull request merged**